Update support for Node 18

shawnmcknight opened the pull request:
This PR augments the `engines` property of `package.json` to support `^18.0.0` as well as `^16.0.0`.  Additionally, it adds a new node 18 configuration.

kthompson23 commented:
Does there need to be a `node18` folder?

kthompson23 reviewed (commented):
Do we need to create a `node18` folder?

shawnmcknight commented:
> Do we need to create a `node18` folder?

I hadn't really considered anything beyond fixing peer dependency warnings, but its probably a good idea.  I added a node18 folder and updated the README.

kthompson23 reviewed (approved):
Looks good.
